zluda/src/impl/dark_api.rs

```diff
@@ -62,6 +62,31 @@ impl CudaDarkApi for CudaDarkApiZluda {
         device::primary_ctx_get(pctx, hip_dev).into_cuda()
     }
 
+    unsafe extern "system" fn set_device_flags(
+        dev: cuda_types::CUdevice,
+        flags: c_uint,
+    ) -> CUresult {
+        use hip_runtime_sys::*;
+        let hip_dev = FromCuda::from_cuda(dev);
+        let mut active_dev = 0i32;
+        hipGetDevice(&mut active_dev as _);
+        if hip_dev != active_dev {
+            hipSetDevice(hip_dev);
```

Comment:
Same here.

Reply:
Aborting is not a bad idea for debug builds. However, the user will expect the application aborts only when a fatal error is occurred. (like memory access violation)
We can return `CUDA_ERROR_INVALID_VALUE`.

Reply:
Or course I meant returning error without going further. Aborting the whole program is one of the worst things external library can do (and AMD sometimes do just that...)